File: data-science-lab-site/data-science-lab-site/Data/Models/PluginPackage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore.Metadata.Internal;

namespace data_science_lab_site.Data.Models
{
    public class PluginPackage
    {
        public int Id { get; set; } 
        [Required, Display(Name = "Package's Name")]
        public string Name { get; set; }
        [Required, Display(Name = "Owner of GitHub Repository"), StringLength(256, MinimumLength = 1, ErrorMessage = "Must be between 1 and 256 characters.")]
        public string Owner { get; set; }
        [Required, Display(Name = "Repository Name"), StringLength(256, MinimumLength = 1, ErrorMessage = "Must be between 1 and 256 characters")]
        public string RepositoryName { get; set; }
        public IList<Plugin> Plugins { get; set; } = new List<Plugin>();
        public string UserId { get; set; }
        public ApplicationUser User { get; set; }
    }
}

File: data-science-lab-app/shared/models/plugin_package.ts
import { Plugin } from './plugin';

export class PluginPackage {
    public name: string;
    public owner: string;
    public repositoryName: string;
    public username: string;
    public plugins: Plugin[];
    public install: boolean;

    constructor(pluginPackage:
        {
            name: string, owner: string, repositoryName: string,
            username: string, plugins?: Plugin[], install?: boolean
        }) {
        this.name = pluginPackage.name;
        this.owner = pluginPackage.owner;
        this.repositoryName = pluginPackage.repositoryName;
        this.username = pluginPackage.username;
        this.plugins = pluginPackage.plugins || [];
        this.install = pluginPackage.install || false;
    }
}
